### PR TITLE
[Enhancement] Get db bingdings of builtin storage volume

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 14 (92.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/SharedNothingStorageVolumeMgr.java|0|1|00.00%|[162]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/SharedNothingStorageVolumeMgr.java [hunk 1]
```diff
@@ -159,7 +158,7 @@ public void validateStorageVolumeConfig() throws InvalidConfException {
     }
 
     @Override
-    protected Set<Long> getTableBindingsOfBuiltinStorageVolume() {
-        return new HashSet<>();
+    protected List<List<Long>> getBindingsOfBuiltinStorageVolume() {
+        return new ArrayList<>();
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/server/SharedNothingStorageVolumeMgr.java:162]: shall provide a list with two valid elements, both elements are empty(), avoid caller NPE when doing get(0), get(1) under any unexpected circumstance

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 14 (92.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/SharedNothingStorageVolumeMgr.java|0|1|00.00%|[163]|
|:large_blue_circle:|com/starrocks/server/SharedDataStorageVolumeMgr.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|3|3|100.00%|[]|

**Review by silverbullet233 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**